pkg/util/log/log.go

```diff
@@ -26,23 +27,35 @@ var (
 	bufferedLogger *dslog.BufferedLogger
 )
 
+type RateLimitedLoggerCfg struct {
+	Enabled            bool
+	LogsPerSecond      float64
+	LogsPerSecondBurst int
+	Registry           prometheus.Registerer
+}
+
 // InitLogger initialises the global gokit logger (util_log.Logger) and overrides the
 // default logger for the server.
-func InitLogger(cfg *server.Config, buffered bool) {
-	l := newBasicLogger(cfg.LogFormat, buffered)
-
-	// when using util_log.Logger, skip 5 stack frames.
-	logger := log.With(l, "caller", log.Caller(5))
-	// Must put the level filter last for efficiency.
-	Logger = level.NewFilter(logger, cfg.LogLevel.Gokit)
+func InitLogger(cfg *server.Config, buffered bool, rateLimitedCfg RateLimitedLoggerCfg) {
```

Comment:
Taking the `server.Config` in input was an hack because we had to both initialise a `log.Interface` logegr and gokit logger. I think now we don't need to take `server.Config` in input anymore, and we should just return the logger instance. The caller should be responsible to set the logger on `server.Config`.